src/hotspot/share/oops/cpCache.cpp

```diff
@@ -266,7 +266,7 @@ void ConstantPoolCacheEntry::set_direct_or_vtable_call(Bytecodes::Code invoke_co
     }
     if (invoke_code == Bytecodes::_invokestatic) {
       assert(method->method_holder()->is_initialized() ||
-             method->method_holder()->is_reentrant_initialization(Thread::current()),
+             method->method_holder()->is_reentrant(Thread::current()),
```

Comment:
Please change this back - `is_reentrant` has no meaning by itself, reentrant with respect to what?

Reply:
I can have a is_reentrant_initialization() and an is_reentrant_linking() method if that makes it clearer.